libqfieldsync/layer.py

```diff
@@ -1016,15 +1016,29 @@ def copy(self, target_path, copied_files, keep_existent=False):
             suffix = uri_parts[1]
 
         if self.is_file:
-            source_path, file_name = os.path.split(self.filename)
-            basename, extensions = get_file_extension_group(file_name)
-            for ext in extensions:
-                dest_file = os.path.join(target_path, basename + ext)
-                if os.path.exists(os.path.join(source_path, basename + ext)) and (
-                    keep_existent is False or not os.path.isfile(dest_file)
-                ):
-                    shutil.copy(os.path.join(source_path, basename + ext), dest_file)
+            if hasattr(QgsFileUtils, "sidecarFilesForPath"):
+                # QGIS >= 3.22
+                files_to_copy = QgsFileUtils.sidecarFilesForPath(self.filename)
+                files_to_copy.add(self.filename)
+                for file_to_copy in files_to_copy:
+                    source_path, file_name = os.path.split(file_to_copy)
```

Comment:
I have slight preference over `pathlib` over `os.path`, but I see the codebase is 50/50 for both approaches, so let's leave it for now.